### DISCUSSION TITLE
Trying to install Material Web Components but never works.

### DISCUSSION THREAD

**Discussion by HaiziIzzudin** [material-components/material-web#5214] [Q&A] [answered]
Hello to y'all. I try to install Material Web Components to be used in my website. But no matter how I install it, it always doesn't work. Tried installing using node.js doesn't work, manually downloading all the modules, doesn't work. Using CDN, doesn't work. Using frameworks like lit.dev works using their playground, but installing locally than use from that doesn't work. I don't know what works. Hopefully you all can help me here. I'm not pro, but I'm learning rn.

OS (1st try): Windows 11 IoT Enterprise
OS (2nd try): Debian Linux via wsl
Node version at time of install: Latest LTS 20.10.0
Code Editor: VS Code
Web Preview: Firefox

Idk how to print verbosely, but what i do know, no elements of MWC is displaying. It just shows normal HTML tag/ text.

Edit: I just realized there is same discussion like mine [discussions/4427](https://github.com/material-components/material-web/discussions/4427). I'll try that first, then I'll let y'all know if it fixes.

**Comment by HaiziIzzudin** [comment]
Nope, it doesn't work.

**Comment by Aworldc** [comment]
when you try installing it via nodejs, what output do you get?

**Reply by HaiziIzzudin** [reply]
Given my OS rn is Windows 11 IoT Enterprise, below is the output.
``` powershell
PS C:\Users\user> node -v
v20.9.0
PS C:\Users\user> npm -v
10.2.2
PS C:\Users\user> cd .\my-website\
PS C:\Users\user\my-website> ls
PS C:\Users\user\my-website> npm install @material/web

up to date, audited 9 packages in 3s

found 0 vulnerabilities
```

**Reply by Aworldc** [reply, marked as answer]
Ok, Npm doesn't seem to be the issue. From the limited information available, it looks like you may be trying to use material web without a build tool.

Try `npm install vite`, create an `index.html` (if you don't have one), then run `npx vite`, which will start up a server.

The thing with material web is you need some sort of build tool that transforms it's code into code understandable by a browser. [Vite](https://vitejs.dev/) is one of those build tools, and it usually just works.

After that, you can just create a JavaScript file, import it in the html (e.g. `<script type="module" src="./main.js">`), and import the components you need in that js file:
```js
import '@material/web/button/elevated-button.js';
import '@material/web/button/filled-button.js';
```

After that, you can just use those components in the html file. (e.g. `<md-filled-button>Hello, world!</md-filled-button>`)

**Reply by HaiziIzzudin** [reply]
Your solution works! Thank you very much. Marked it as answer. I will later publish my own ways on using MWC within my repository. Thank you again!

**Reply by asyncliz** [reply]
Our quick start guide also has a [section on building](https://github.com/material-components/material-web/blob/main/docs/quick-start.md#building)! Those docs use rollup as an example bundler.

Any javascript bundlers should work, including Vite and Webpack!

**Comment by datvm** [comment]
Does this work for you? https://github.com/datvm/mdc-start-template

Just clone that one, run `npm run build` then serve it with any server of your choice (I heard Node devs use `http-server` but I myself use IIS from VS Code).

**Reply by HaiziIzzudin** [reply]
Hi, thanks for the reply. Unfortunately I only know javascript (not typescript). Very sorry :(
